src/main/java/net/mcreator/ui/procedure/RetvalProcedureSelector.java

```diff
@@ -0,0 +1,251 @@
+					procedureNameString);
+
+			if (procedureNameString != null) {
+				ModElement element = new ModElement(mcreator.getWorkspace(), procedureNameString,
+						ModElementType.PROCEDURE);
+				ModElementGUI<?> newGUI = ModElementType.PROCEDURE.getModElementGUI(mcreator, element, false);
+				if (newGUI != null) {
+					newGUI.showView();
+					newGUI.setModElementCreatedListener(generatableElement -> {
+						String modName = JavaConventions.convertToValidClassName(
+								generatableElement.getModElement().getName());
+						refreshList();
+						setSelectedProcedure(modName);
+					});
+					mcreator.getApplication().getAnalytics().async(() -> mcreator.getApplication().getAnalytics()
+							.trackEvent(AnalyticsConstants.EVENT_NEW_MOD_ELEMENT,
+									ModElementType.PROCEDURE.getReadableName(), null, null));
+				}
+			}
+		});
+
+		edit.setMargin(new Insets(0, 0, 0, 0));
+		edit.setOpaque(false);
+		edit.setContentAreaFilled(false);
+		edit.addActionListener(e -> {
+			if (getSelectedProcedure() != null) {
+				ModElement selectedProcedureAsModElement = mcreator.getWorkspace()
+						.getModElementByName(getSelectedProcedure().getName());
+				ModElementGUI<?> modeditor = selectedProcedureAsModElement.getType()
+						.getModElementGUI(mcreator, selectedProcedureAsModElement, true);
+				if (modeditor != null)
+					modeditor.showView();
+			}
+		});
+
+		if (fixedValue != null)
+			add("Center", PanelUtils.westAndEastElement(
+					PanelUtils.westAndCenterElement(PanelUtils.totalCenterInPanel(procedures), fixedValue),
+					PanelUtils.totalCenterInPanel(PanelUtils.gridElements(1, 2, add, edit))));
+		else
+			add("East", PanelUtils.totalCenterInPanel(
+					PanelUtils.join(FlowLayout.LEFT, 0, 1, procedures, new JEmptyBox(1, 1),
+							PanelUtils.gridElements(1, 2, add, edit))));
+
+		add("West", PanelUtils.join(FlowLayout.LEFT, 4, 4, top));
+
+		procedures.setToolTipText(L10N.t("action.procedure.match_dependencies"));
+
+		procedures.setPrototypeDisplayValue(new CBoxEntry("XXXXXXXXXX", null));
+
+		if (fixedValue != null && width != 0)
+			fixedValue.setPreferredSize(new Dimension(width, 0));
+
+		GeneratorConfiguration gc = mcreator.getGeneratorConfiguration();
+		if (gc.getGeneratorStats().getModElementTypeCoverageInfo().get(ModElementType.PROCEDURE)
+				== GeneratorStats.CoverageStatus.NONE)
+			setEnabled(false);
+	}
+
+	@Override protected CBoxEntry updateDepsList(boolean smallIcons) {
+		CBoxEntry selected = super.updateDepsList(true);
+
+		edit.setEnabled(selected != null && !selected.string.equals(defaultName));
+
+		if (fixedValue != null)
+			fixedValue.setEnabled(!edit.isEnabled());
+
+		return selected;
+	}
+
+	@Override public void setEnabled(boolean enabled) {
+		if (fixedValue != null)
+			fixedValue.setEnabled(enabled);
+
+		if (enabled) {
+			setBackground((Color) UIManager.get("MCreatorLAF.LIGHT_ACCENT"));
+		} else {
+			setBackground((Color) UIManager.get("MCreatorLAF.DARK_ACCENT"));
```

Comment:
Maybe you can inline this condition as it's not quite simple.

Reply:
Long line then, not really needed